### ISSUE TITLE
sqlite3-execute crashes when called with bounds argument without callback argument.

### ISSUE THREAD

**Issue by ikazuhiro** [syohex/emacs-sqlite3#4]
Thank you for your work!

But current `sqlite3-execute` function  crashes Emacs when called with `bounds` argument without `callback` argument.  In this case, `sqlite3-core-execute` is called with `bounds` and `callback` argument and `callbacck`'s content is the same with `bounds`.

**Comment by syohex** [comment]
Could you show me code which reproduces this issue ? I cannot reproduce.

**Comment by ikazuhiro** [comment]
I noticed the crash is a special case, normally the bug raises an error.  Here is the minimal recipe.

```lisp
(let (db)
  (require 'sqlite3)
  (setq db (sqlite3-new))
  (sqlite3-execute-batch db "CREATE TABLE foo(id integer primary key, editor text);")
  (sqlite3-execute-batch db "INSERT INTO foo(editor) values(?)" ["Emacs"])
  (sqlite3-execute db  "SELECT * FROM foo WHERE editor = ?;" ["Emacs"]))
```

Backrace is the below.

```
Debugger entered--Lisp error: (invalid-function ["Emacs"])
  internal--module-call(#<save-value <pointer 0000000002617650> <unused> <unused> <unused>> #<user-ptr ptr=0000000003637140 finalizer=0000000063041810> "SELECT * FROM foo WHERE editor = ?;" ["Emacs"] ["Emacs"])
  apply(internal--module-call #<save-value <pointer 0000000002617650> <unused> <unused> <unused>> (#<user-ptr ptr=0000000003637140 finalizer=0000000063041810> "SELECT * FROM foo WHERE editor = ?;" ["Emacs"] ["Emacs"]))
  sqlite3-core-execute(#<user-ptr ptr=0000000003637140 finalizer=0000000063041810> "SELECT * FROM foo WHERE editor = ?;" ["Emacs"] ["Emacs"])
  (let* ((rargs (reverse args)) (callback (car rargs)) bounds) (if (functionp callback) (progn (setq rargs (cdr rargs)))) (if rargs (progn (setq bounds (car rargs)))) (if (and bounds (not (vectorp bounds))) (progn (setq bounds (vconcat bounds)))) (sqlite3-core-execute sqlite query bounds callback))
  sqlite3-execute(#<user-ptr ptr=0000000003637140 finalizer=0000000063041810> "SELECT * FROM foo WHERE editor = ?;" ["Emacs"])
  (let ((db (sqlite3-new))) (require (quote sqlite3)) (sqlite3-execute-batch db "CREATE TABLE foo(id integer primary key, editor text);") (sqlite3-execute-batch db "INSERT INTO foo(editor) values(?)" ["Emacs"]) (sqlite3-execute db "SELECT * FROM foo WHERE editor = ?;" ["Emacs"]))
  eval((let ((db (sqlite3-new))) (require (quote sqlite3)) (sqlite3-execute-batch db "CREATE TABLE foo(id integer primary key, editor text);") (sqlite3-execute-batch db "INSERT INTO foo(editor) values(?)" ["Emacs"]) (sqlite3-execute db "SELECT * FROM foo WHERE editor = ?;" ["Emacs"])) nil)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

**Comment by syohex** [comment]
I have fixed this issue by #5. Please check latest version.
Thanks for reporting issue.

**Comment by ikazuhiro** [comment]
I confirmed the problem was fixed.  Thank you.
